### ISSUE TITLE
Warning with --set-baseline

### ISSUE THREAD

**Issue by sebastianbergmann** [vimeo/psalm#2108]
```
./tools/psalm --version                                                                                    
Psalm 3.4.12@86e5e50c1bb492045e70f2ebe1da3cad06e4e9b2
```
```
./tools/psalm --config=.psalm/config.xml --set-baseline=.psalm/baseline.xml --no-progress --show-info=false
Writing error baseline to file...
Baseline saved to .psalm/baseline.xml.PHP Warning:  file_get_contents(): Filename cannot be empty in phar:///usr/local/src/phpunit/tools/psalm/src/psalm.php on line 363

Warning: file_get_contents(): Filename cannot be empty in phar:///usr/local/src/phpunit/tools/psalm/src/psalm.php on line 363
PHP Warning:  file_put_contents(): Filename cannot be empty in phar:///usr/local/src/phpunit/tools/psalm/src/psalm.php on line 376

Warning: file_put_contents(): Filename cannot be empty in phar:///usr/local/src/phpunit/tools/psalm/src/psalm.php on line 376


------------------------------
No errors found!
------------------------------

Checks took 5.96 seconds and used 184.205MB of memory
Psalm was able to infer types for 90.8352% of the codebase
```
AFAICS, `.psalm/baseline.xml` is generated correctly.
